### PR TITLE
17wf/testsMarketplace

### PR DESCRIPTION
**Pull request by Piwakk**

### CODE REVIEW

File: associations/views/marketplace.py [hunk 1]
```diff
@@ -1,147 +1,202 @@
...
     queryset = Product.objects.all()
     serializer_class = ProductSerializer
-    permission_classes = (CanManageMarketplace,)
+    permission_classes = (IfMarketplaceAdminThenCRUDElseR,
+                          IfMarketplaceEnabledThenCRUDElseRAndMarketplaceAdminOnlyCRUD,)
+
     filter_backends = (filters.SearchFilter,)
     search_fields = ("name",)
 
+    def get_queryset(self):
```
Comment (abocquet) [inline associations/views/marketplace.py]: This function has no body

Comment (Piwakk) [reply associations/views/marketplace.py]: That's because it's nobody ~

Comment (Piwakk) [reply associations/views/marketplace.py]: Actually it has a body:
```python
"""The user has access to the products coming from every enabled marketplace and to the products of every
        marketplace she is a library administrator of."""

        # The library for which the user is library administrator.
        marketplaces = [role.association.marketplace for role in self.request.user.roles.all() if role.marketplace]

        return Product.objects.filter(Q(marketplace__enabled=True) | Q(marketplace__in=marketplaces))
```

Comment (Piwakk) [reply associations/views/marketplace.py]: Actually it does have a body:
```python
"""The user has access to the products coming from every enabled marketplace and to the products of every
        marketplace she is a library administrator of."""

        # The library for which the user is library administrator.
        marketplaces = [role.association.marketplace for role in self.request.user.roles.all() if role.marketplace]

        return Product.objects.filter(Q(marketplace__enabled=True) | Q(marketplace__in=marketplaces))
```

**Pull request merged**